lightning/src/chain/keysinterface.rs

```diff
@@ -848,7 +883,27 @@ impl KeysManager {
 	/// Note that until the 0.1 release there is no guarantee of backward compatibility between
 	/// versions. Once the library is more fully supported, the docs will be updated to include a
 	/// detailed description of the guarantee.
+	///
+	/// This method cannot be used for nodes that wish to support receiving multi-node or phantom
+	/// payments; [`KeysManager::new_multi_receive`] must be used instead.
+	///
+	/// Switching between this method and [`KeysManager::new_multi_receive`] will invalidate any
+	/// previously issued invoices and attempts to pay previous invoices will fail.
 	pub fn new(seed: &[u8; 32], starting_time_secs: u64, starting_time_nanos: u32) -> Self {
+		Self::new_inner(seed, starting_time_secs, starting_time_nanos, None)
+	}
+
+	/// Similar to [`KeysManager::new`], but allows the node using this `KeysManager` to receive
+	/// phantom node payments. `cross_node_seed` must be the same across all phantom-receiving nodes
+	/// and also the same across restarts, or else payments may fail.
+	///
+	/// Switching between this method and [`KeysManager::new`] will invalidate any
+	/// previously issued invoices and attempts to pay previous invoices will fail.
+	pub fn new_multi_receive(seed: &[u8; 32], starting_time_secs: u64, starting_time_nanos: u32, cross_node_seed: &[u8; 32]) -> Self {
+		Self::new_inner(seed, starting_time_secs, starting_time_nanos, Some(cross_node_seed))
+	}
+
+	fn new_inner(seed: &[u8; 32], starting_time_secs: u64, starting_time_nanos: u32, cross_node_seed: Option<&[u8; 32]>) -> Self {
```

Comment:
As a thought experiment related to my other comment, but going a different direction: Why make `cross_node_seed` an `Option` at all? If it were required as a "shared seed", then you remove the problem entirely. Not saying this is necessarily the right approach. But curious as to why it might be the wrong approach. I suppose because it would break older versions? If we were designing this from scratch, would we require it?

Reply:
It's only useful for phantom payments, and indeed it does break older versions' inbound payments. Not saying that would have been the wrong approach if we were designing from scratch, but could maybe be argued that it's an extra/confusing argument for users to whom the feature is not relevant